### PR TITLE
feat/DocxWriter Table class

### PR DESCRIPTION
**Pull request by younesdessia**
* **What kind of changes does this PR introduce?** 
- [ ] Bug fix
- [X] new features
- [X] performance 
- [ ] docs update

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
- [ ] Yes:
- [ ] No

* **Other information**:

### CODE REVIEW

File: code_pylint.py [hunk 1]
```diff
@@ -54,7 +54,7 @@
                                 'too-many-return-statements', 'import-outside-toplevel',
                                 'too-many-instance-attributes', 'broad-except', 'bare-except', "broad-exception-caught",
                                 'too-many-public-methods', 'too-many-statements', 'undefined-loop-variable',
-                                'attribute-defined-outside-init']
+                                'attribute-defined-outside-init', 'wrong-spelling-in-docstring']
```
Comment (GhislainJ) [inline code_pylint.py:57]: Wrong spelling in docstring should not be added to ERRORS_WITHOUT_TIME_DECREASE as the aim of it is to lower the number of spelling errors over time...

File: dessia_common/document_generator.py [hunk 2]
```diff
@@ -108,6 +108,35 @@ def from_markdown(cls, markdown_text: str):
...
+
+    def __init__(self, rows: List[List[str]]):
+        self.rows = rows
+
+    def add_to_document(self, document: docx.Document, style="TableGrid"):
+        """
+        Add the table to the document.
+
+        :param document: The document to which the table will be added.
+        :type document: docx.Document
```
Comment (GhislainJ) [inline dessia_common/document_generator.py:122]: Useless redundant type documenting (already set in method signature)

File: dessia_common/document_generator.py [hunk 3]
```diff
@@ -108,6 +108,35 @@ def from_markdown(cls, markdown_text: str):
...
+        :param style: The table style to use. Default is "TableGrid". Other available styles include:
+                        - "LightShading": adds light shading to the table cells.
+                        - "LightGrid": similar to "TableGrid", but with lighter grid lines.
+                        - "MediumShading1", "MediumShading2": adds medium-level shading to the table cells,
+                        with different levels of intensity.
+                        - "DarkList": displays the table as a numbered list, with a dark background color for each item.
+                        - "ColorfulGrid": uses bright, contrasting colors for the table borders and cell backgrounds.
+                        - "LightList": displays the table as a bulleted list, with a light background color for each
+                        item.
+        :type style: str
```
Comment (GhislainJ) [inline dessia_common/document_generator.py:132]: Could set this in method signature

File: dessia_common/optimization.py [hunk 4]
```diff
@@ -91,7 +91,7 @@ def get_model_from_vector(self):
 
 
 class InstantiatingModelOptimizer(Optimizer):
-    """ Abstract class, to be subclassed by real class. Instantiate a new model at each point request. """
+    """ Abstract class, to be sub-classed by real class. Instantiate a new model at each point request. """
```
Comment (GhislainJ) [inline dessia_common/optimization.py:94]: Should be "inherited"

**Review by GhislainJ — CHANGES_REQUESTED** [review]
We agreed last week that there wasn't supposed to be further devs on this feature. Has this been prioritized ?

File: code_pylint.py (hunk 1 above)
Comment (younesdessia) [reply code_pylint.py]: ok,  I changed that

File: dessia_common/document_generator.py (hunk 3 above)
Comment (younesdessia) [reply dessia_common/document_generator.py:132]: same thing !

File: dessia_common/document_generator.py (hunk 2 above)
Comment (younesdessia) [reply dessia_common/document_generator.py:122]: I didn't completely understand !

File: dessia_common/optimization.py (hunk 4 above)
Comment (younesdessia) [reply dessia_common/optimization.py]: ok !

File: dessia_common/document_generator.py [hunk 5]
```diff
@@ -148,6 +175,7 @@ def __init__(self, paragraphs: List[Paragraph] = None, section: Section = None,
         if paragraphs is None:
             paragraphs = []
         self.paragraphs = paragraphs
+        self.table = table
```
Comment (GhislainJ) [inline dessia_common/document_generator.py:178]: After another look at the code, I find it weird that DocxWriter always have a table attribute. What if some document doesn't or has many tables ?

File: .pyenchant_dessia.dict [hunk 6]
```diff
@@ -53,3 +53,6 @@ uuid
 uuids
 visjs
 volmdlr
+Dessia's
```
Comment (GhislainJ) [inline .pyenchant_dessia.dict:56]: This document should be kept alphabetically ordered

**Review by GhislainJ — CHANGES_REQUESTED** [review]

File: dessia_common/document_generator.py (hunk 5 above)
Comment (younesdessia) [reply dessia_common/document_generator.py:178]: I added a list

File: .pyenchant_dessia.dict (hunk 6 above)
Comment (younesdessia) [reply .pyenchant_dessia.dict]: it's done

File: dessia_common/document_generator.py (hunk 5 above)
Comment (younesdessia) [reply dessia_common/document_generator.py:178]: by default table is None

**Comment by younesdessia** [comment]
@GhislainJ conflicts fixed

**Review by GhislainJ — APPROVED** [review]

**Pull request merged**